File: BikeShop.Backend/BikeShop.Products/BikeShop.Products.Domain/Entities/ProductTag.cs
using System.Text.Json.Serialization;

namespace BikeShop.Products.Domain.Entities;

public class ProductTag : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public int ParentId { get; set; }
    public bool IsCollapsed { get; set; }
    public bool IsRetailVisible { get; set; }
    public bool IsB2BVisible { get; set; }
    public bool IsUniversal { get; set; }
    public int SortOrder { get; set; }
    
    [JsonIgnore] public IList<TagToProductBind> TagToProductBinds { get; set; }
}

File: BikeShop.Frontend/webclient/src/entities/entities/ProductTag.ts
export interface ProductTag {
    id: string
    name: string
    parentId: string

    isCollapsed: boolean
    isRetailVisible: boolean
    isB2BVisible: boolean
    isUniversal: boolean
    sortOrder: number

    createdAt: string
    updatedAt: string
    enabled: boolean
}